### DISCUSSION TITLE
How to convert jpeg format to RAW ppm

### DISCUSSION THREAD

**Discussion by dosshra** [ImageMagick/ImageMagick#2096] [Help] [answered]
Hello
I am trying to convert multiple jpeg files to ppm RAW format by using the code `mogrify -format ppm *.jpeg`. But it seems to generate ASCII files. How do I convert to ppm RAW files?
Thank you

**Comment by fmw42** [comment, marked as answer]
What is your IM version and platform?  When I do that on my Mac OSX Sierra with IM 6.9.11.16 Q16 it produces RAW files unless I add -compress none to make it ASCII.  For example, take a 10x10 section of the rose: image and make it PNG

`convert rose:[10x10+0+0] rose.png`

Now test:

`mogrify -format ppm rose.png`
`cat rose.ppm`

```
P6
10 10
255
0/-20.62/83.:3-92-80-91.80-80-/.,0/-50-71-81,80,6/,81-81-81.,,,,-+0/,1.+2,(2-*0,+2,+2++2,+,-/+-/.--/-)/*'-+(,**,**.++.++,/2+/2..0-.--+),*++)+('(**+,*+.0.//-/1.21.30-3/,2-*-*+,+,*)-23-34-53.961961:2.80-3.,300-.0782782994<:5@=8B:8A95=73<75:548939:4>>8A?:D?:FA<EA:E@:G?8E=4671681==8B@;FA?HC>HC>JD<MB7J=0
```
<br>

**So this is raw data. P6 is raw and the data is not ASCII**


Now if I add **-compress none**, I get

`mogrify -format ppm -compress none rose.png`
`cat rose.ppm`

```
P3
10 10
255
48 47 45 50 48 46 54 50 47 56 51 46 58 51 45 57 50 45 56 48 45 57 49 46 56 48 45 56 48 45 
47 46 44 48 47 45 53 48 45 55 49 45 56 49 44 56 48 44 54 47 44 56 49 45 56 49 45 56 49 46 
44 44 44 44 45 43 48 47 44 49 46 43 50 44 40 50 45 42 48 44 43 50 44 43 50 43 43 50 44 43 
44 45 47 43 45 47 46 45 45 47 45 41 47 42 39 45 43 40 44 42 42 44 42 42 46 43 43 46 43 43 
44 47 50 43 47 50 46 46 48 45 46 45 45 43 41 44 42 43 43 41 43 40 39 40 42 42 43 44 42 43 
46 48 46 47 47 45 47 49 46 50 49 46 51 48 45 51 47 44 50 45 42 45 42 43 44 43 44 42 41 45 
50 51 45 51 52 45 53 51 46 57 54 49 57 54 49 58 50 46 56 48 45 51 46 44 51 48 48 45 46 48 
55 56 50 55 56 50 57 57 52 60 58 53 64 61 56 66 58 56 65 57 53 61 55 51 60 55 53 58 53 52 
56 57 51 57 58 52 62 62 56 65 63 58 68 63 58 70 65 60 69 65 58 69 64 58 71 63 56 69 61 52 
54 55 49 54 56 49 61 61 56 66 64 59 70 65 63 72 67 62 72 67 62 74 68 60 77 66 55 74 61 48 
```
<br>

**P3 is Ascii format PPM and the data is indeed ASCII.**

**Reply by dosshra** [reply]
I use ImageMagick 6.9.7-4 Q16 x86_64 on Ubuntu 18.04. Using cat  I can see the P6 in the header. 
It seems that my problem is not the ASCII format.
Thank you.
